src/bp/ui-studio/src/web/components/Shared/Interface/index.tsx

```diff
@@ -0,0 +1,185 @@
+      <div className={classnames(style.container, { [style.sidebar_hidden]: !sidebarVisible })}>
+        <SplitPane split="vertical" defaultSize={width} size={sidebarVisible ? width : 0}>
+          {props.children}
+        </SplitPane>
+      </div>
+    </HotKeys>
+  )
+}
+
+export const Section = (props: SectionProps) => {
+  const [isOpen, setOpen] = useState(!props.collapsed)
+
+  return (
+    <React.Fragment>
+      <div className={style.sidebar_section} onClick={() => setOpen(!isOpen)}>
+        <strong>
+          {!props.hideCaret && <Icon icon={isOpen ? 'caret-down' : 'caret-right'} />}
+          {props.label || ''}
+        </strong>
+        <ButtonGroup minimal={true} onClick={e => e.stopPropagation()}>
+          {props.actions && props.actions.map((action, idx) => SectionAction(action, idx))}
+        </ButtonGroup>
+      </div>
+      <Collapse isOpen={isOpen}>{props.children}</Collapse>
+    </React.Fragment>
+  )
+}
+
+export const SearchBar = (props: SearchBarProps) => {
+  const [text, setText] = useState('')
+  const handleTextChanged = e => {
+    setText(e.target.value)
+    props.onChange && props.onChange(e.target.value)
+  }
+
+  return (
+    <div className={style.searchBar}>
+      <ControlGroup fill={true}>
+        <InputGroup placeholder={props.placeholder || 'Search'} value={text} onChange={handleTextChanged} />
+        <Button icon="search" className={Classes.FIXED} onClick={e => props.onClick && props.onClick(e)} />
+      </ControlGroup>
+    </div>
+  )
+}
+
+export const ItemList = (props: ItemListProps) => {
+  return (
+    <div className={style.itemList}>
+      <ul>
+        {props.items &&
+          props.items.map((item, idx) => (
+            <li key={idx}>
+              <div style={{ width: '80%' }} onClick={() => props.onElementClicked && props.onElementClicked(item)}>
+                {item.label}
+              </div>
+              <div style={{ marginLeft: 'auto ' }}>
+                {props.actions &&
+                  props.actions.map(action => (
+                    <Tooltip key={idx + action.tooltip} content={action.tooltip} position={Position.RIGHT}>
+                      <Icon icon={action.icon} onClick={() => action.onClick && action.onClick(item)} />
+                    </Tooltip>
+                  ))}
+              </div>
+            </li>
+          ))}
+      </ul>
+    </div>
+  )
+}
+
+export const PaddedContent = props => <div style={{ padding: '5px' }}>{props.children}</div>
+export const Sidebar = props => <div className={style.sidebar}>{props.children}</div>
+
+export const KeyboardShortcut = (props: KeyboardShortcutsProps) => {
+  const ACTION_KEY = navigator.platform.toUpperCase().indexOf('MAC') >= 0 ? 'cmd' : 'ctrl'
+  return (
+    <p>
+      {props.label || ''}
+      &nbsp;
+      {props.keys &&
+        props.keys.map((key, idx) => {
+          const realKey = key === 'ACTION' ? ACTION_KEY : key
+          return idx > 0 ? (
+            <span key={idx}>
```

Comment:
giving the index as key for react might end up with degraded performance, it's the same as not giving any key at all. Try to give it a uniq key